packages/modeling/src/operations/booleans/martinez/index.js

```diff
@@ -76,14 +77,32 @@ const compareBBoxes = (subject, clipping, sbbox, cbbox, operation) => {
  * Convert from geom2 to martinez data structure
  */
 const toMartinez = (geometry) => {
-  const outlines = geom2.toOutlines(geometry)
-  outlines.forEach((outline) => {
+  const outlines = []
+  geom2.toOutlines(geometry).forEach((outline) => {
     // Martinez expects first point == last point
-    outline.push(outline[0])
+    if (vec2.equals(outline[0], outline[outline.length - 1])) {
+      outlines.push(outline)
+    } else {
+      outlines.push([...outline, outline[0]])
+    }
   })
   return [outlines]
 }
 
+/*
+ * Convert martinez data structure to geom2
+ */
+const fromOutlines = (outlines) => {
```

Comment:
Just asking... but is there an issue with geom2 if the first point == last point?

Reply:
There is no "issue". But martinez always produces outlines with first == last. So if you run a bunch of boolean operations in sequence, you can end up with many duplicate points at the end of the array. Since JSCAD does not require first == last, I remove the duplicates in order to produce the cleanest, simplest geometry that we can.

Reply:
@platypii cool. let's make sure the documentation mentions that each outline is closed between first and last points.